### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by vchinnakotla**

### CODE REVIEW

File: data.tf [hunk 1]
```diff
@@ -0,0 +1,15 @@
+data "aws_iam_policy_document" "route53_query_logs" {
+  statement {
+    actions = [
+      "logs:CreateLogStream",
+      "logs:PutLogEvents",
+    ]
+
+    resources = ["arn:aws:logs:*:*:log-group:/aws/route53/*"]
```
Comment (joseph-wortmann) [inline data.tf]: Please change the resource to point _directly_ to the log-group that you created. This policy is too broad...

File: cloudwatch.tf [hunk 2]
```diff
@@ -0,0 +1,9 @@
+resource "aws_cloudwatch_log_resource_policy" "route53_query_logging_policy" {
+  policy_document = "${data.aws_iam_policy_document.route53_query_logs.json}"
+  policy_name     = "route53-query-logging-policy"
+}
+
+resource "aws_cloudwatch_log_group" "query_log" {
+  name              = "/aws/route53/${var.hosted_zone_id}"
```
Comment (joseph-wortmann) [inline cloudwatch.tf]: Is there a reason that we are using the hosted_zone_id and not the name of the hosted zone to name the log group? I think that the latter would be easier to read...

**Review by joseph-wortmann — CHANGES_REQUESTED** [review]
You need to add an `outputs.tf` and a `locals.tf` file to this.

In the `outputs.tf` please output relevant information about the log group and the query log that you create (arn, etc.).

File: variables.tf [hunk 3]
```diff
@@ -1,3 +1,8 @@
+variable "domain_name" {
```
Comment (joseph-wortmann) [inline variables.tf:1]: Don't make them pass this in. Use the aws_route53_hosted_zone data resource to get it instead.

File: data.tf [hunk 4]
```diff
@@ -5,7 +5,7 @@ data "aws_iam_policy_document" "route53_query_logs" {
       "logs:PutLogEvents",
     ]
 
-    resources = ["arn:aws:logs:*:*:log-group:/aws/route53/*"]
+    resources = ["arn:aws:logs:*:*:log-group:/aws/route53/${var.domain_name}/*"]
```
Comment (joseph-wortmann) [inline data.tf]: Can't you get the actual arn from the log group you create in cloudwatch.tf?

File: variables.tf [hunk 5]
```diff
@@ -0,0 +1,10 @@
+variable "hosted_zone_id" {
+  description = "Hosted zone id"
+  type        = "string"
+}
+
+variable "log_retention_duration" {
+  description = "Cloudwatch Log retention in days, defaults to 30"
+  default     = 30
```
Comment (joseph-wortmann) [inline variables.tf]: Make this 7 days

**Comment by vchinnakotla** [comment]
Changes are made, please review.

**Review by joseph-wortmann — APPROVED** [review]

**Pull request merged**